classes/local/v1p2/oneroster_client.php

```diff
@@ -0,0 +1,1001 @@
+<?php
```

Comment:
All of this code duplication can be avoided using inheritance. Happy to chat it through in the meeting tomorrow